### PR TITLE
PT-361 complex ownership search request implementation

### PR DESCRIPTION
**Pull request by zolt**

### CODE REVIEW

File: api/src/main/kotlin/com/rarible/protocol/union/api/controller/OwnershipController.kt [hunk 1]
```diff
@@ -4,26 +4,25 @@ import com.rarible.core.logging.Logger
...
 import com.rarible.protocol.union.dto.parser.IdParser
 import com.rarible.protocol.union.dto.parser.OwnershipIdParser
-import kotlinx.coroutines.ExperimentalCoroutinesApi
-import org.slf4j.LoggerFactory
 import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.RestController
 
 @RestController
 class OwnershipController(
-    private val ownershipSourceSelectService: OwnershipSourceSelectService,
```
[the remark this answers is not in the document]
Comment (zolt) [reply api/src/main/kotlin/com/rarible/protocol/union/api/controller/OwnershipController.kt]: auto-formatting, fixed

File: enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/search/internal/EsOwnershipQueryBuilderService.kt [hunk 2]
```diff
@@ -41,4 +46,38 @@ class EsOwnershipQueryBuilderService(
         mustMatchTerm(filter.owner.fullId(), EsOwnership::owner.name)
         mustMatchTerms(filter.blockchains?.toSet().orEmpty(), EsOwnership::blockchain.name)
     }
+
+    private fun BoolQueryBuilder.applySearchFilter(filter: EsOwnershipsSearchFilter) {
```
[the remark this answers is not in the document]
Comment (zolt) [reply enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/search/internal/EsOwnershipQueryBuilderService.kt:50]: added integration test

File: api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OwnershipElasticService.kt [hunk 3]
```diff
@@ -66,4 +70,44 @@ class OwnershipElasticService(
             }
         )
     }
+
+    override suspend fun search(request: OwnershipSearchRequestDto): OwnershipsDto {
```
[the remark this answers is not in the document]
Comment (zolt) [reply api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/OwnershipElasticService.kt:74]: added integration test

File: api/src/test/kotlin/com/rarible/protocol/union/api/service/elastic/OwnershipElasticServiceIt.kt [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+            coEvery {
+                enrichmentAuctionService.fetchAuctionsIfAbsent(any(), any())
+            } returns emptyMap()
+        }
+
+    }
+
+
+    @Test
+    internal fun `search by request test`() {
```
Comment (augur) [inline api/src/test/kotlin/com/rarible/protocol/union/api/service/elastic/OwnershipElasticServiceIt.kt:159]: Were all filter cases tested?

Comment (zolt) [reply api/src/test/kotlin/com/rarible/protocol/union/api/service/elastic/OwnershipElasticServiceIt.kt:161]: add more test cases

**Review by augur — APPROVED** [review]

**Pull request merged**